### PR TITLE
w252659874 Procfile for Worker Instances

### PR DESCRIPTION
**Pull request by whatisdot**
Closes https://www.wrike.com/open.htm?id=252659874

# Features

## Update Run Requirements
Multicontainer utilizes ECS, which enables us to easily get environment details from the ECS Task Definitions. This means we no longer need to execute commands via SSM, increasing reliability by eliminating text parsing for environment details and polling for the result of remote command execution.

## Updated Environment Types
Each process in a `Procfile` will be given its own elastic beanstalk environment. Each environment will enable scaling a single container-type independent of the others.

## Multi-deploy
This will deploy the image to each environment type, specifying the command that should execute via a configuration file (Dockerrun.aws.json).

## Expectations
* All environments must be created ahead of time.
* All environments must have their environment variables updated independently.
* This requires 

# Testing
* Check out this branch from git-scripts.
* Checkout the [snapsheet-url-shortener](https://github.com/bodyshopbidsdotcom/snapsheet-url-shortener) and switch to the branch `dockerize`.
* Push to the stage environment `git_push_eb stage`
* Try to run a command that connects to the dataase: `git_run_eb stage rails c`
* [Check in ECS](https://us-west-2.console.aws.amazon.com/ecs/home?region=us-west-2#/clusters) to see that worker environments are running.

screen shot 2018-07-26 at 2 50 03 pm

### CODE REVIEW

**Comment by whatisdot** [comment]
This is dependent on a [PR in `infra`](https://github.com/bodyshopbidsdotcom/infra/pull/15).

File: lib/push_eb.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    return self.start_process([
+      'eb', 'deploy',
+      self.project_name()+'-'+self.args.environment,
+      '--message', '"'+version+'"'
+    ])
+
+  def clean(self):
+    self.start_process(['rm', '-rf', '.elasticbeanstalk'])
+
+  def dockerrun_config(self, version, config_file='Procfile'):
```
Comment (jeet4320) [inline lib/push_eb.py]: `config_file` is not used in the function

File: lib/push_eb.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    with open(".elasticbeanstalk/config.yml", "w") as config_file:
+      yaml.dump(config_settings, config_file, default_flow_style=False)
+
+  def build(self):
+    tag_name = self.project_name()+':'+self.version
+    self.start_process(['docker', 'build', '-t', tag_name, '.'])
+    self.start_process(['docker', 'tag', tag_name, self.ecr_image_uri()])
+    runfile = open('Dockerrun.aws.json', 'w')
+    runfile.write(json.dumps(self.dockerrun_config(), indent=2, sort_keys=True))
+    runfile.close()
```
Comment (whatisdot) [inline lib/push_eb.py]: This is where `Dockerrun.aws.json` gets created. We will later remove it after the `eb deploy` command uploads it to S3.

**Comment by jeet4320** [comment]
@whatisdot change profile `assumed_role_config` with `user_session` before merging as this PR is merged https://github.com/bodyshopbidsdotcom/git-scripts/pull/340

**Comment by whatisdot** [comment]
I discovered an issue where requests to the load balancer don't make it to the web server. All things look operational, but it is taking longer than expected to identify and fix the issue.

I'm going to break out the non-proc related changes into a separate PR, and putting "Not Ready" on this until the configuration issue can be resolved.

**Pull request merged**